src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -142,15 +161,70 @@ protected ExtensionsRunner(Extension extension) throws IOException {
     }
 
     /**
-     * This method is call from {@link ExtensionsInitRequestHandler}.
+     * Marks the extension initialized.
+     */
+    public void setInitialized() {
+        this.initialized = true;
```

Comment:
Do we really need this boolean value?

Reply:
I think it's of value for an extension to know if it's received all the data it needs.  For example, you need the NamedXContent in order to successfully create a detector.    You don't know you have that without checking this boolean.

Reply:
A-OK with boolean here.